pkg/scheduler/placement/fixed_rule.go

```diff
@@ -121,12 +121,19 @@ func (fr *fixedRule) placeApplication(app *objects.Application, queueFn func(str
 		if parentName == "" {
 			parentName = configs.RootQueue
 		}
-		queueName = parentName + configs.DOT + fr.queue
+		childQueueName := replaceDot(fr.queue)
+		if err = configs.IsQueueNameValid(childQueueName); err != nil {
+			return "", err
+		}
+		queueName = parentName + configs.DOT + childQueueName
```

Comment:
This breaks a use case. I can set the fixed part to a partial path and not qualify it.
So I can set "prod.high" as the fixed part, not qualified. Use a parent rule of `User` to finally create a queue `root.wilfred.prod.high`, after this change it would create `root.wilfred.prod_dot_high`

The fixed part compliance for all parts of the possible unqualified path should be added to the `initialise()` code

Reply:
Yes, we should not replace dot in this rule. https://issues.apache.org/jira/browse/YUNIKORN-2660 has been filed to make the changes like you explained.